### PR TITLE
Fix test for _cat/nodeattrs

### LINKED ISSUE #12558: cat.nodeattrs/10_basic/Test somtimes fails
FAILURE 0.14s | RestIT.test {p0=cat.nodeattrs/10_basic/Test cat nodes attrs output} <<<
   > Throwable #1: java.lang.AssertionError: field [$body] was expected to match the provided regex but didn't
   > Expected: ((\S+)\s+(\S+)\s+(\d{1,3}\.){3}\d{1,3}\s+(\S+)\s+(\S+)\s*)+

See http://build-us-00.elastic.co/job/es_core_master_regression/2933

### PR DESCRIPTION
**Pull request by nik9000**
Closes #12558

### CODE REVIEW

**Comment by dakrone** [comment]
Doesn't this make the test worthless since it could then not output any attributes and the test would still pass?

**Comment by clintongormley** [comment]
One alternative is to require test nodes to have some node attr in place. We already have other requirements for the REST tests like setting `path.repo` etc, so there is precedent.

Also, we have "skip-feature" functionality in the REST tests which will only run a test if some condition is met.  So we could check if any node attribute is set, then only run the test if that is the case.

**Comment by nik9000** [comment]
> One alternative is to require test nodes to have some node attr in place. We already have other requirements for the REST tests like setting path.repo etc, so there is precedent.

I'll just do this. I was trying to find a simple solution to this while working on something else, just stealing time while something more important was running tests. I'll test it with a specific nodeattr in place.

**Comment by dakrone** [comment]
That sounds like a better alternative to me too

**Comment by nik9000** [comment]
> That sounds like a better alternative to me too

Ok - I think I got it.

**Comment by clintongormley** [comment]
@nik9000 as a side note (ie for a separate change) it would be great to document the settings that are required to run the REST tests successfully.  This info will be important for all the clients too.

**Comment by nik9000** [comment]
> @nik9000 as a side note (ie for a separate change) it would be great to document the settings that are required to run the REST tests successfully. This info will be important for all the clients too.

Yeah - that's a good idea. And I'm not truly sure what that list is. `EsRestTestCase` is probably the right place to look though!

**Comment by nik9000** [comment]
@dakrone could you have another look at this? I think its no longer useless.

File: dev-tools/Elasticsearch.launch [hunk 1]
```diff
@@ -14,5 +14,5 @@
 <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.elasticsearch.bootstrap.Elasticsearch"/>
 <stringAttribute key="org.eclipse.jdt.launching.PROGRAM_ARGUMENTS" value="start"/>
 <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="elasticsearch"/>
-<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Xms256m -Xmx1g -Djava.awt.headless=true -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=75 -XX:+UseCMSInitiatingOccupancyOnly -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=logs/heapdump.hprof -Delasticsearch -Des.foreground=yes -ea -Des.path.home=target/eclipse_run -Des.security.manager.enabled=false"/>
+<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Xms256m -Xmx1g -Djava.awt.headless=true -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=75 -XX:+UseCMSInitiatingOccupancyOnly -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=logs/heapdump.hprof -Delasticsearch -Des.foreground=yes -ea -Des.path.home=target/eclipse_run -Des.security.manager.enabled=false&#10;-Des.node.attr=test"/>
```
Comment (dakrone) [inline dev-tools/Elasticsearch.launch]: `&#10;-Des.node.attr=test` ? I think the `&#10` is an error?

Comment (nik9000) [inline dev-tools/Elasticsearch.launch]: Looks like `&#10;` a line feed. I hit enter in the box when I was changing this. It works and its easier to read. I have no idea why its just a line feed in the xml though.

**Comment by nik9000** [comment]
@dakrone, I believe this is waiting on a review from you.

**Comment by dakrone** [comment]
LGTM

**Comment by nik9000** [comment]
Squashed and rebased. Retested and all passes. Merging to master and 2.x.

**Comment by nik9000** [comment]
All done.

**Pull request merged**